### ISSUE TITLE
Log Formatting

### ISSUE THREAD

**Issue by Knight-Yoshi** [yiisoft/yii2#13855]
Log formatting should be able to be changed easily by setting a property in the config file.
Additionally it should accept Mapped Diagnostic Constant like what Log4PHP and Log4J have. So that other data can be added to the left of the message.

Using Log4PHP this is the pattern we use
`'[%-5.5p] [%d{H:i:s}] [%X{username}] [%-15.15logger] [%-3.3line] | %message - %relative(ms)%newline'`

Which is the log type (info, debug, error, fatal, warn), timestamp, MDC username, the logger name (always 15 characters), the line it was called on, and the message.

Also as a side note if a debug log type could be added that would be appreciated as well.

**Comment by marekpetras** [comment]
would be great to have actual memory consumption as well

**Comment by cebe** [comment]
What exactly is missing from the functionality provided by `yii\log\Target::$prefix` ? http://www.yiiframework.com/doc-2.0/yii-log-target.html#$prefix-detail

**Comment by samdark** [comment]
Format could be used to do postfix as well. Prefix can't.

**Comment by cebe** [comment]
So this is about introducing a configuration/callback that allows customizing `formatMessage()` right?

**Comment by samdark** [comment]
Yes. There's a format for config that is already used for Log4PHP and Log4J so we can reuse it.
